sklearn/impute.py

```diff
@@ -942,10 +985,10 @@ def transform(self, X):
 
         X, Xt, mask_missing_values = self._initial_imputation(X)
 
-        if self.n_iter == 0:
+        if self.n_iter_ == 0:
```

Comment:
This is also not tested. Could we force the `n_iter_` and check that we have `X == Xt`.